src/math/gaussian_quadrature.rs

```diff
@@ -0,0 +1,46 @@
+// Gaussian Quadrature Module
+//https://en.wikipedia.org/wiki/Gaussian_quadrature
+
+pub fn gaussian_quadrature(a: f64, b: f64, f: impl Fn(f64) -> f64, num_points: usize) -> f64 {
+    let (points, weights) = get_gaussian_quadrature_points_weights(num_points);
+
+    let mut result = 0.0;
+
+    for i in 0..num_points {
+        let x_i = 0.5 * (a + b) + 0.5 * (b - a) * points[i];
+        result += weights[i] * f(x_i);
+    }
+
+    0.5 * (b - a) * result
+}
+
+fn get_gaussian_quadrature_points_weights(num_points: usize) -> (Vec<f64>, Vec<f64>) {
+    // Hardcoded values for Gaussian Quadrature points and weights
+    match num_points {
+        1 => (vec![0.0], vec![2.0]),
+        2 => (vec![-1.0 / 3.0, 1.0 / 3.0], vec![1.0, 1.0]),
+        3 => (
+            vec![-0.7745966692414834, 0.0, 0.7745966692414834],
+            vec![0.5555555555555556, 0.8888888888888888, 0.5555555555555556],
+        ),
+        _ => unimplemented!("Gaussian Quadrature not implemented for this number of points"),
+    }
+}

```

Comment:
I would highly suggest to generalize it for any _number of points_/_order_. It also seems that it uses Legendre polynomials. It also could be more general.

Reply:
Hmm,i am kinda cofused here ,can you please tell me more information about what i should add?

Reply:
Let's start with basic stuff: it would be great to call `gaussian_quadrature` with `order` being `20` and obtain a valid result.

Reply:
Ahh, i see what you mean,sorry for being late ,been kinda busy,thanks for the explanation my man.

Reply:
i keep trying to add the options but the result just keeps getting more complicated and wrong ,my purpose in here wat to give an example of another numerical approach to integrals,after some research 3 is the most command order for this method (and most accurate) ,mind if i leave it like this?

Reply:
I think we should go for the general case.